mlir/lib/Dialect/Vector/Transforms/VectorEmulateNarrowType.cpp

```diff
@@ -129,6 +132,45 @@ static FailureOr<Operation *> getCompressedMaskOp(OpBuilder &rewriter,
       auto denseAttr = DenseElementsAttr::get(newMaskType, newMaskValues);
       newMask = rewriter.create<arith::ConstantOp>(loc, newMaskType, denseAttr);
     }
+  } else if (constantOp) {
+    assert(shape.size() == 1 && "expected 1-D mask");
```

Comment:
updated `shape` to `maskShape` to better represent the meaning.

Currently limiting it to only 1 d because the `arith.constant` op needs different treatment from the other two ops.

Reply:
I honestly don't know if there is any > 1-d masks being generated, but `arith.constant` does allow arbitrary dimension constants being generated in this way. Is there any mechanism guarding it off from any such cases before this pass?

cc @hanhanW @dcaballe

Reply:
I don't really know the answer.. My intuition is no in IREE project. My understanding is that all the n-D vector.transfer_read/write ops are unrolled to 1-D vector loads/stores on CPU path before we reach out to the emulation. GPU side, we don't use masking in IREE now.

However, I remember that @dcaballe added some n-D emulation support few months ago, so I'm not sure now.

I'd suggest to move this check before "TypeSwitch" and see what's happening. If it does not break anything, we don't need to make `arith.constant` case special. Otherwise, we leave it to where it is and add a TODO.

What do you think?

Reply:
> I'd suggest to move this check before "TypeSwitch" and see what's happening. If it does not break anything, we don't need to make arith.constant case special. 

+1

No need to handle complexities that are not required. And if they do become required then at least we'll have a good motivating example to inform the design :)

Reply:
Turns out it breaks things as some masks indeed created by other kinds of ops do have multiple dimensions. I am leaving it as is with a TODO for now. I will come back later.